pandas/io/parsers.py

```diff
@@ -227,14 +227,19 @@
     Note that the entire file is read into a single DataFrame regardless,
     use the `chunksize` or `iterator` parameter to return the data in chunks.
     (Only valid with C parser)
+buffer_lines : int, default None
```

Comment:
it *seems* its not respected at all now. so we should just remove this argument (or raise if its not ``None``)